src/ucp/rndv/rndv.c

```diff
@@ -1243,10 +1269,13 @@ UCS_PROFILE_FUNC(ucs_status_t, ucp_rndv_rts_handler,
     ucp_worker_h worker         = arg;
     ucp_rndv_rts_hdr_t *rts_hdr = data;
 
-    /* RNDV is supported with TAG protocol only */
-    ucs_assert(rts_hdr->flags & UCP_RNDV_RTS_FLAG_TAG);
+    if (rts_hdr->flags & UCP_RNDV_RTS_FLAG_TAG) {
+        return ucp_tag_rndv_process_rts(worker, rts_hdr, length, tl_flags);
+    }
+
+    ucs_assert(rts_hdr->flags & UCP_RNDV_RTS_FLAG_AM);
 
-    return ucp_tag_rndv_process_rts(worker, rts_hdr, length, tl_flags);
+    return ucp_am_rndv_process_rts(arg, data, length, tl_flags);
```

Comment:
if (rts_hdr->flags & UCP_RNDV_RTS_FLAG_TAG) {
    return ucp_tag_rndv_process_rts(worker, rts_hdr, length, tl_flags);
} else {
   ucs_assert(rts_hdr->flags & UCP_RNDV_RTS_FLAG_AM);
  return ucp_am_rndv_process_rts(worker, rts_hdr, length, tl_flags);
}

Reply:
why? typically we do not do `else`s if we have `return`

Reply:
but both options are equal in "importance", so if/else would be better